staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -154,10 +154,10 @@ func (i *indexedWatchers) terminateAll(objectType reflect.Type, done func(*cache
 		klog.Warningf("Terminating all watchers from cacher %v", objectType)
 	}
 	i.allWatchers.terminateAll(done)
-	for index, watchers := range i.valueWatchers {
+	for _, watchers := range i.valueWatchers {
 		watchers.terminateAll(done)
-		delete(i.valueWatchers, index)
 	}
+	i.valueWatchers = map[string]watchersMap{}
```

Comment:
Deleting changes potentially the amount of allocations in the future.
But terminateAll is supposed to happen rarely enough, that it shouldn't matter that much.